src/applications/drydock/blueprint/DrydockWorkingCopyBlueprintImplementation.php

```diff
@@ -521,116 +522,109 @@ private function applyMerge(
     }
 
     // TM CHANGES BEGIN
-    // We try to rebase the diff off the branch we are trying land on.
-    // this means we can drop any commits which have already been landed.
-    $tmp_branch_name = sprintf(
-      '%s-%d',
-      self::TMP_BRANCH_PREFIX,
-      rand());
+    // We first try to squash merge then if that fails we try to rebase the diff off the branch
+    // we are trying land on. Rebasing means we can drop any commits which have already been landed.
     try {
+      // NOTE: This can never actually generate a commit because we pass
+      // "--squash", but git sometimes runs code to check that a username and
+      // email are configured anyway.
       $interface->execx(
-        'git checkout -b %s %s',
-        $tmp_branch_name,
+        'git -c user.name=%s -c user.email=%s merge --no-stat --squash -- %s',
+        'drydock',
+        'drydock@phabricator',
         $src_ref);
     } catch (CommandException $ex) {
-      $display_command = csprintf(
-        'git checkout -b %s %R',
-        $tmp_branch_name,
-        $src_ref);
-      $error = DrydockCommandError::newFromCommandException($ex)
-        ->setPhase(self::PHASE_SQUASHMERGE)
-        ->setDisplayCommand($display_command);
-      $lease->setAttribute('workingcopy.vcs.error', $error->toDictionary());
-
-      throw $ex;
-    }
-
-    try {
-      $interface->execx('git rebase -');
-    } catch (CommandException $ex) {
-
-      // If we fail to rebase we insted do the original squash merge without rebase
+      $tmp_branch_name = sprintf(
+        '%s-%d',
+        self::TMP_BRANCH_PREFIX,
+        rand());
       try {
         $interface->execx(
-          'git rebase --abort && git checkout - && git -c user.name=%s -c user.email=%s merge --no-stat --squash -- %s',
-          'drydock',
-          'drydock@phabricator',
+          'git reset HEAD --hard && git checkout -b %s %s',
```

Comment:
Is the reset just precautionary? I think I'm confused about what branch we're on here

Reply:
its because we might have failed to squash merge in the previous step but there might be changes in the working copy as part of the merge operation so we reset to HEAD before attempting the rebase flow. I'll add some comments to this part to clarify.